drivers/wifi/esp_at/esp.c

```diff
@@ -825,7 +825,23 @@ static int cmd_ipd_parse_hdr(struct esp_data *dev,
 		char *remote_ip;
 		long port;
 
-		err = esp_pull_quoted(&str, str_end, &remote_ip);
+#if defined(CONFIG_WIFI_ESP_AT_FETCH_VERSION)
+		if (dev->at_version.major >= 2) {
+			/* Quoted per AT version 2.1.0/2.2.0 */
+			err = esp_pull_quoted(&str, str_end, &remote_ip);
+		} else {
+			/* NOT quoted per AT version 1.7.0 */
+			err = esp_pull_raw(&str, str_end, &remote_ip);
+		}
+#else
+		if (!IS_ENABLED(CONFIG_WIFI_ESP_AT_VERSION_1_7)) {
```

Comment:
I think we should just use `CONFIG_WIFI_ESP_AT_VERSION_1_7`, so build time information, without relying on runtime detection. It won't make sense to run `CONFIG_WIFI_ESP_AT_VERSION_1_7` with AT 2.x version or vice-versa, as there are plenty of differences in AT commands.

Reply:
Agreed and removed runtime version check